api/overview/usage_examples.adoc

```diff
@@ -0,0 +1,479 @@
+* <<ruby,Ruby>>
+  - <<ruby-example1,Example 1: A simple **GET** with Ruby>>
+  - <<ruby-example2,Example 2: Using **POST** to edit a group with Ruby>>
+  - <<ruby-example3,Example 3: A **GET** using filtering with Ruby>>
+* <<python,Python>>
+  - <<python-example1,Example 1: A simple **GET** with Python>>
+  - <<python-example2,Example 2: Using **POST** to edit a group with Python>>
+  - <<python-example3,Example 3: A **GET** using filtering with Python>>
+
+'''''
+
+[[curl]]
+== cURL
+
+The cURL command can be used from the shell. However the shell's interpretation of some characters, for example **&**, can make it awkward to use for complex cases.
+To help alleviate this the payload can be passed to cURL using the **--data** flag, where the payload is stored in a file in the format specified 
+in the content-type, **JSON** in our example.
+
+The JSON returned is formated in the examples using **python -m json.tool**
+
+[[curl-example1]]
+==== Example 1: A simple **GET** with cURL
+
+Get the collection **groups**
+
+----
+curl --user admin:smartvm --insecure --request GET https://192.0.2.0/api/groups | python -m json.tool
+----
+
+===== Example 1: Result
+
+Result truncated and showing only the returned **resources**.
+
+----
+{
+    ...
+    "resources": [
+        {
+            "href": "https://192.0.2.0/api/groups/2"
+        },
+        {
+            "href": "https://192.0.2.0/api/groups/3"
+        },
+        ...
+        {
+            "href": "https://192.0.2.0/api/groups/23"
+        }
+    ],
+    ...
+}
+----
+
+[[curl-example2]]
+==== Example 2: Using **POST** to edit a group with cURL
+
+Edit an instance of the collection **groups**
+
+In this example the payload **data** is read from a file.
+
+----
+$ cat request.json
+{
+  "action" : "edit",
+  "resource" : {
+    "description" : "updated group description from cURL"
+  }
+}
+----
+
+Use the data in file request.json to edit an instance of the collection **groups**
+
+----
+curl --data @request.json \
+     --header "Content-Type: application/json" \
+     --user admin:smartvm \
+     --insecure \
+     --request POST https://192.0.2.0/api/groups/23 | python -m json.tool
+----
+
+===== Example 2: Result
+
+----
+{
+    "created_on": "2019-04-15T19:49:42Z",
+    "description": "updated group description from cURL",
+    "group_type": "user",
+    "href": "https://192.0.2.0/api/groups/23",
+    "id": "23",
+    "sequence": 22,
+    "settings": null,
+    "tenant_id": "2",
+    "updated_on": "2019-06-19T19:15:51Z"
+}
+----
+
+[[curl-example3]]
+==== Example 3: A **GET** using filtering with cURL
+
+Filtering can be done by passing arguments in the URL. Arguments are preceded by the **&** character, which can cause problems as the shell interprets the **&** character as an instruction to: **run this command in the background**. To avoid this the arguments can instead be passed to curl using the **--get** flag combined with each argument being passed with the **--data** flags.
```

Comment:
replace arguments/Arguments with parameters/Parameters  (this is more accurate as these are called query parameters).

Reply:
👍 Will do.